### PR TITLE
Ensure we only ask for a symbol's documentation comment once 

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Right now computing hover information is a fairly expensive portion of an LSIF dump. One small hotspot was we were fetching documentation comment information at least twice: once for the documentation itself and once for computing the exception info. This ensures we fetch it just once per generation.

This also generally cleans up the AbstractSymbolDescriptionBuilder:

1. Duplicate local functions for returns/value is unified.
2. Removes a bunch of specific logic in the builder which supported finding the correct documentation in favor of the helper we already had.

To support the second one, I added some methods to DocumentationComment to return a DocumentationComment with the contents of a <param> tag as the summary text for a specific parameter; this allows everything else to be able to operate as if that's a regular summary tag and removes some special cases in the code.

This by itself isn't the hugest speedup (maybe a couple of percent) but generally makes the code easier to follow and easier to see where the remaining expensive bits are in traces.

### CODE REVIEW

File: SpellingExclusions.dic
Comment (jasonmalinowski) [inline SpellingExclusions.dic:1]: VS 17.6 fixed the issues around .dic files having to be UTF-16 encoded, so we can just do this now.

File: src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs [hunk 1]
```diff
@@ -151,48 +144,31 @@ protected SemanticModel GetSemanticModel(SyntaxTree tree)
...
                 AddCaptures(firstSymbol);
 
-                AddDocumentationContent(firstSymbol);
+                AddDocumentationContent(firstSymbol, firstSymbolDocumentationComment);
             }
 
-            private void AddDocumentationContent(ISymbol symbol)
+            private void AddDocumentationContent(ISymbol symbol, DocumentationComment documentationComment)
             {
-                var formatter = Services.GetRequiredLanguageService<IDocumentationCommentFormattingService>(_semanticModel.Language);
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs:172]: This logic was alreayd duplicated in ISymbolExtensions_2.cs, so it's unified now.

File: src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.cs [hunk 2]
```diff
@@ -2,28 +2,22 @@
...
-using Microsoft.CodeAnalysis.Classification.Classifiers;
 using Microsoft.CodeAnalysis.Host;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageService
 {
     internal abstract partial class AbstractSymbolDisplayService : ISymbolDisplayService
     {
-        protected readonly LanguageServices Services;
-        protected readonly IStructuralTypeDisplayService AnonymousTypeDisplayService;
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.cs:21]: For some reason we were passing around both LanguageServices and also one instance of a specific language service, so cleaned up.

File: src/Workspaces/Core/Portable/Shared/Utilities/DocumentationComment.cs [hunk 3]
```diff
@@ -328,6 +328,14 @@ private void ParseCallback(XmlReader reader)
             return text;
         }
 
+        public DocumentationComment? GetParameter(string parameterName)
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Shared/Utilities/DocumentationComment.cs:331]: I should add some better doc comments here -- but to simplify handling in the other parts of the code, the idea here is if we have a <params> member, we can create a new DocumentationComment object that has a <summary> tag with the same contents, so everything else can treat that identically.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**